crates/cxx-qt-lib-extras/src/assertion_utils.h

```diff
@@ -3,52 +3,9 @@
 // clang-format on
 // SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
 // SPDX-FileContributor: Leon Matthes <leon.matthes@kdab.com>
+// SPDX-FileContributor: Matt Aber <matt.aber@kdab.com>
 //
 // SPDX-License-Identifier: MIT OR Apache-2.0
 #pragma once
 
-#include <array>
-#include <cassert>
-#include <cstdint>
-
-namespace rust::cxxqtlib1::assertion_utils {
-template<typename iter>
-constexpr static ::std::size_t
-calc_align_size(const iter rbegin,
-                const iter rend,
-                const ::std::size_t actual_alignment)
-{
-  ::std::size_t rows = 0;
-  ::std::size_t accum = 0;
-
-  for (auto it = rbegin; it != rend; ++it) {
-    assert(*it <= actual_alignment);
-
-    if (it + 1 != rend) {
-      if (accum + *it == actual_alignment) {
-        accum += *it;
-      } else {
-        if (accum + *it + *(it + 1) <= actual_alignment)
-          accum += *it;
-        else
-          accum = actual_alignment;
-      }
-      if (accum == actual_alignment) {
-        accum = 0;
-        ++rows;
-      }
-    } else {
-      ++rows;
-    }
-  }
-
-  return rows * actual_alignment;
-}
-} // namespace rust::cxxqtlib1::assertion_utils
-
-#define assert_alignment_and_size(TYPE, EXP_ALIGN, ARR, ARR_SZ)                \
-  static_assert(EXP_ALIGN == alignof(TYPE));                                   \
-  static_assert(                                                               \
-    ::rust::cxxqtlib1::assertion_utils::calc_align_size<                       \
-      ::std::array<::std::size_t, ARR_SZ>::const_reverse_iterator>(            \
-      ::std::rbegin(ARR), ::std::rend(ARR), alignof(TYPE)) == sizeof(TYPE));
+#include "cxx-qt-lib/src/assertion_utils.h"
```

Comment:
seems this fails? Wonder if we aren't bundling the assertion_utils.h header into the cxx-qt-lib export?

Reply:
@mattkdab i think you can put it back to `#include <cxx-qt-lib/assertion_utils.h>` then what you need to do is

- in `crates/cxx-qt-lib/` move the assertion_utils.h from the src/ folder to the include/ folder as we export the include folder here https://github.com/KDAB/cxx-qt/blob/86d00e56a431192410a0ba19936c745f8040962a/crates/cxx-qt-lib/build.rs#L295
- add `assertion_utils.h` to the write_headers like we do for common here https://github.com/KDAB/cxx-qt/blob/86d00e56a431192410a0ba19936c745f8040962a/crates/cxx-qt-lib/build.rs#L50
- remove `assertion_utils` from the manual rerun-if at the end https://github.com/KDAB/cxx-qt/blob/86d00e56a431192410a0ba19936c745f8040962a/crates/cxx-qt-lib/build.rs#L337
- update any includes of it in cxx-qt-lib .cpp files to use the new `cxx-qt-lib/assertion_utils.h` path rather than the relative `../assertion_utils.h`